lib/client/conntest/connection_tester.go

```diff
@@ -22,16 +22,48 @@ import (
 
 	"github.com/gravitational/trace"
 
+	"github.com/gravitational/teleport/api/client/proto"
 	"github.com/gravitational/teleport/api/defaults"
 	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/lib/auth"
+	wanlib "github.com/gravitational/teleport/lib/auth/webauthn"
 )
 
+type MfaResponse struct {
+	// TotpCode is a code for a otp device.
+	TotpCode string `json:"totpCode"`
+	// WebauthnResponse is a response from a webauthn device.
+	WebauthnResponse *wanlib.CredentialAssertionResponse `json:"webauthnResponse"`
+}
+
+func (r *MfaResponse) getMFAResponseProtoReq() (*proto.MFAAuthenticateResponse, error) {
+	if r.TotpCode != "" && r.WebauthnResponse != nil {
+		return nil, trace.BadParameter("only one MFA response field can be set")
+	}
+
+	if r.TotpCode != "" {
+		return &proto.MFAAuthenticateResponse{Response: &proto.MFAAuthenticateResponse_TOTP{
+			TOTP: &proto.TOTPResponse{Code: r.TotpCode},
+		}}, nil
+	}
+
+	if r.WebauthnResponse != nil {
+		return &proto.MFAAuthenticateResponse{Response: &proto.MFAAuthenticateResponse_Webauthn{
+			Webauthn: wanlib.CredentialAssertionResponseToProto(r.WebauthnResponse),
+		}}, nil
+	}
+
+	// Can be nil since it's optional.
```

Comment:
I renamed the function instead to `getOptionalMFAProtoRequest` since we moved it out and made it public. Because maybe future uses we would require a response?